client/my-sites/domains/domain-management/name-servers/custom-nameservers-form.jsx

```diff
@@ -94,12 +93,24 @@ class CustomNameserversForm extends React.PureComponent {
 		} );
 	}
 
+	removeNameserver( index ) {
+		const nameservers = [ ...this.props.nameservers ];
+		nameservers.splice( index, 1 );
+		return nameservers;
+	}
+
+	editNameserver( index, nameserver ) {
+		const nameservers = [ ...this.props.nameservers ];
+		nameservers[ index ] = nameserver;
+		return nameservers;
+	}
```

Comment:
Nice idea 👍  

I implemented your suggestions in a35a8a2c515eacd1549ba0596b25e9aefa801587 however I had to revert the `handleChange` one in 89f6b26f6f170812f17e357fea39df1504e1e50a because of the way the custom nameservers form is architected - `.map()` won't work because we're also using this function to create array items, not just to change them. To repro, change again to `map` and you'll see that your edits in the custom nameservers rows don't get reflected in the UI.

Reply:
Maybe we should start using some immutable helper library like Immer to make our reducers more resilient and elegant 🙂